### PR TITLE
[MINI-5694] Launch MiniApps when offline

### PR DESCRIPTION
**Pull request by laubengaier**
# Description
MiniApps didn't open when running the device in offline mode.
Fixing this by intercepting the error from sdk side and launch the MiniApp with `fromCache = true`.
Fix to `MiniAppWithTermsViewModel` for toggling between online and offline mode

**Video (including mode switch fix)**
https://rak.box.com/s/lp9gcibx1bxqe0rl1kh6g08wcnu0dkok

## Links
https://jira.rakuten-it.com/jira/browse/MINI-5694

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

**Review by GondiTilak — APPROVED** [review]
Looks good to me :+1:

File: Sources/Classes/ui/MiniAppSUIView.swift [hunk 1]
```diff
@@ -8,26 +8,30 @@ public struct MiniAppSUIView: UIViewRepresentable {
     @ObservedObject var handler: MiniAppSUIViewHandler
 
     var params: MiniAppViewParameters
+    var fromCache: Bool
```
Comment (Climbatize) [inline Sources/Classes/ui/MiniAppSUIView.swift:11]:
```suggestion
    var fromCache: Bool = false
```

File: Sources/Classes/ui/MiniAppSUIView.swift [hunk 2]
```diff
@@ -8,26 +8,30 @@ public struct MiniAppSUIView: UIViewRepresentable {
...
-    public init(params: MiniAppViewParameters.DefaultParams, handler: MiniAppSUIViewHandler) {
+    public init(params: MiniAppViewParameters.DefaultParams, fromCache: Bool = false, handler: MiniAppSUIViewHandler) {
         self.params = .default(params)
+        self.fromCache = fromCache
         self.handler = handler
     }
 
     public init(urlParams: MiniAppViewParameters.UrlParams) {
         self.params = .url(urlParams)
+        self.fromCache = false
```
Comment (Climbatize) [inline Sources/Classes/ui/MiniAppSUIView.swift:21]:
```suggestion
```

**Review by Climbatize — APPROVED** [review]
LGTM

File: Example/Controllers/SwiftUI/Features/Terms/MiniAppWithTermsViewModel.swift [hunk 3]
```diff
@@ -108,4 +115,46 @@ class MiniAppWithTermsViewModel: ObservableObject {
...
         }
     }
+
+    func isOffline(error: Error) -> Bool {
+        let error = error as NSError
+        if let maSdkError = error as? MASDKError {
+            return maSdkError.code == -1020 || maSdkError.code == -1009
+        }
+        return [NSURLErrorNotConnectedToInternet, NSURLErrorTimedOut, NSURLErrorDataNotAllowed].contains(error.code)
+    }
```
[the remark this answers is not in the document]
Comment (laubengaier) [reply Example/Controllers/SwiftUI/Features/Terms/MiniAppWithTermsViewModel.swift]: 👍 updated, the error list was missing `NSURLErrorDataNotAllowed` to make it work added it

**Pull request merged**